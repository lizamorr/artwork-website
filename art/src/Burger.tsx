import React, { useState } from 'react';
import Menu from './Menu';

const Burger = () => {
    const [isOpen, setIsOpen] = useState(false);
    return (
        <div onClick={() => setIsOpen(!isOpen)} className={isOpen ? "close" : "burger"} aria-label="Toggle menu">
            <span />
            <span />
            <span />
            {isOpen && <Menu />}
        </div>
    )
}

export default Burger;