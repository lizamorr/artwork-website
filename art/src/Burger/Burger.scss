.burger {
  position: absolute;
  top: 1.5%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 1.5rem;
  width: 1.5rem;
  cursor: pointer;
  z-index: 10;
  margin: 10px;
  right: 0.5rem;

  span {
    height: 0.13rem;
    background: black;
    border-radius: 4px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;
  }
}

.close {
  position: absolute;
  top: 1.5%;
  margin: 10px;
  right: 0.5rem;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 1.5rem;
  width: 1.5rem;
  cursor: pointer;
  padding: 0;
  z-index: 10;

  span {
    height: 0.13rem;
    background: black;
    border-radius: 4px;
    transition: all 0.3s linear;
    position: relative;
    transform-origin: 1px;
  }
  span:nth-of-type(1) {
    transform: rotate(45deg);
  }
  span:nth-of-type(2) {
    display: none;
  }
  span:nth-of-type(3) {
    transform: rotate(-45deg);
    margin-top: 6px;
  }
}

@media only screen and (max-width: 405px) {
  .burger,
  .close {
    transform: scale(0.7);
  }
}
