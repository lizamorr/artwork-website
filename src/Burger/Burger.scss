.menu-icon {
  position: absolute;
  top: 0.1rem;
  cursor: pointer;
  z-index: 10;
  margin: 10px;
  padding: 0;
  border-radius: 9px;
  right: 0.1rem;
  opacity: 1 !important;
}
