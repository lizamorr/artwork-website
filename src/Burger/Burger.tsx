import React, { useState } from 'react';
import Menu from './Menu';

const Burger: React.FC = () => {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <div
      onClick={() => setIsOpen(!isOpen)}
      className={isOpen ? 'close' : 'burger'}
      aria-label="Toggle menu"
    >
      <span />
      <span />
      <span />
      {isOpen && <Menu />}
    </div>
  );
};

export default Burger;
