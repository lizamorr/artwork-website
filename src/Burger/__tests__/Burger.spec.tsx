import Burger from '../Burger';
import renderer from 'react-test-renderer';
import React from 'react';

describe('Burger Component', () => {
  it('should render correctly', () => {
    const component = renderer.create(<Burger />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
