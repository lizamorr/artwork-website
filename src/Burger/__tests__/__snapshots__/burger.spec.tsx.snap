// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Burger Component should render correctly 1`] = `
<div
  aria-label="Toggle menu"
  className="burger"
  onClick={[Function]}
>
  <span />
  <span />
  <span />
</div>
`;
